src/theory/ff/util.h

```diff
@@ -0,0 +1,85 @@
+/******************************************************************************
+ * Top contributors (to current version):
+ *   Alex Ozdemir
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2023 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * utilities
+ */
+
+#include "cvc5_private.h"
+
+#ifndef CVC5__THEORY__FF__UTIL_H
+#define CVC5__THEORY__FF__UTIL_H
+
+// external includes
+#ifdef CVC5_USE_COCOA
+#include <CoCoA/ring.H>
+#endif /* CVC5_USE_COCOA */
+
+// std includes
+#include <unordered_map>
+
+// internal includes
+#include "expr/node.h"
+#include "util/finite_field_value.h"
+
+namespace cvc5::internal {
+namespace theory {
+namespace ff {
+
+/** A finite field model */
+using FfModel = std::unordered_map<Node, FiniteFieldValue>;
+
+/** The result of a subsolver. */
+enum class Result
```

Comment:
Rather than define this, you could be using the `Result` class in `src/util/result.h`, right? It has more than just the enum, but maybe you'd need to define something like that anyway?

Reply:
I don't need anything more than the enum class for this line of PRs, but if you think it'd be good to use the existing `Result` anyways, I'm happy to do that.

Reply:
I see. I think I'd prefer the existing `Result` to be reused anyway since it does not really have much dependencies.